protocol/x/clob/e2e/permissioned_keys_test.go

```diff
@@ -0,0 +1,561 @@
+			},
+		},
+		"Txn rejected by all of authenticator": {
+			smartAccountEnabled: true,
+			blocks: []TestBlockWithMsgs{
+				{
+					Block: 2,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: &aptypes.MsgAddAuthenticator{
+								Sender:            constants.BobAccAddress.String(),
+								AuthenticatorType: "AllOf",
+								Data:              compositeAuthenticatorConfig,
+							},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        300_000,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{1},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+				{
+					Block: 4,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: clobtypes.NewMsgPlaceOrder(
+								testapp.MustScaleOrder(
+									constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20,
+									testapp.DefaultGenesis(),
+								),
+							),
+							Authenticators: []uint64{0},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        0,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{1},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: aptypes.ErrAllOfVerification.ABCICode(),
+							ExpectedLog:      aptypes.ErrAllOfVerification.Error(),
+						},
+					},
+				},
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20.OrderId: false,
+			},
+		},
+		"Txn rejected by any of authenticator": {
+			smartAccountEnabled: true,
+			blocks: []TestBlockWithMsgs{
+				{
+					Block: 2,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: &aptypes.MsgAddAuthenticator{
+								Sender:            constants.BobAccAddress.String(),
+								AuthenticatorType: "AnyOf",
+								Data:              compositeAuthenticatorConfig,
+							},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        300_000,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{1},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+				{
+					Block: 4,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: clobtypes.NewMsgPlaceOrder(
+								testapp.MustScaleOrder(
+									constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20,
+									testapp.DefaultGenesis(),
+								),
+							),
+							Authenticators: []uint64{0},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        0,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{1},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: aptypes.ErrAnyOfVerification.ABCICode(),
+							ExpectedLog:      aptypes.ErrAnyOfVerification.Error(),
+						},
+					},
+				},
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20.OrderId: false,
+			},
+		},
```

Comment:
I think (?) we support multi-msg transaction + authenticator. Should we add a test where one of the messages were not authenticated successfully?

Reply:
Added!